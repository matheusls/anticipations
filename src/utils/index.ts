export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};
