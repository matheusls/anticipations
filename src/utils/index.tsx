import { toast, TypeOptions } from 'react-toastify';

export const centsToReal = (cents: number) => {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL',
  }).format(cents / 100);
};

export const realToCents = (real: string) => {
  return Number(real.replace(/\D+/g, ''));
};

type ErrorToastProps = {
  text: string;
  type: TypeOptions;
};

export const renderErrorToast = (error = '') => {
  const errorToastProps: ErrorToastProps = {
    '500': {
      text: 'Oops, tivemos um problema. Por favor, tente novamente.',
      type: 'error' as const,
    },
    '408': {
      text: 'Oops, nossos servidores estão demorando para responder...',
      type: 'warning' as const,
    },
    offline: {
      text: 'Você parece estar com problemas de conexão!',
      type: 'warning' as const,
    },
  }[error] || {
    text: 'Oops, houve um problema. Por favor, tente novamente.',
    type: 'error' as const,
  };

  const { text, type } = errorToastProps;

  return toast(text, {
    position: 'top-center',
    autoClose: false,
    closeOnClick: true,
    type,
  });
};
