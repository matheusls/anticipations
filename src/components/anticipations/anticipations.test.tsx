import { render, screen } from 'tests/utils';

import Anticipations from './anticipations';
import { centsToReal } from 'utils';

const anticipations = Object.entries({
  '15': 13536,
  '30': 13824,
  '90': 14400,
});

describe('<Anticipations />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <Anticipations anticipations={anticipations} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Anticipations correctly', () => {
    render(<Anticipations anticipations={anticipations} />);

    const title = screen.getByRole('heading', {
      name: /você receberá:/i,
      level: 2,
    });
    const list = screen.getByRole('list');
    const listItems = screen.getAllByRole('listitem');

    expect(title).toBeInTheDocument();
    expect(list).toBeInTheDocument();
    expect(listItems).toHaveLength(3);

    listItems.forEach((item, i) => {
      const [day, value] = anticipations[i];

      expect(item).toHaveTextContent(`Em ${day} dias: ${centsToReal(value)}`);
    });
  });

  it('should render Anticipations with LoadingDots correctly', () => {
    render(<Anticipations anticipations={anticipations} isLoading />);

    expect(screen.getAllByTestId('loading-dots')).toHaveLength(4);
  });
});
