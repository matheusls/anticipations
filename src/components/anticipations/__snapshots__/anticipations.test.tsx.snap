// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anticipations /> should match snapshot 1`] = `
.c1 {
  color: #3d75bb;
  display: block;
  font-size: 1rem;
  font-style: italic;
  font-weight: 700;
  line-height: 1.5;
  margin-bottom: 22px;
  text-transform: uppercase;
  width: 100%;
}

.c1::after {
  background-color: #5d9bec;
  content: '';
  display: block;
  height: 1px;
  margin-top: 4px;
  mix-blend-mode: normal;
  opacity: 0.3;
  width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(209,220,227,0.18);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 35px;
}

.c2 {
  list-style: none;
  margin: 0;
  padding: 0;
  width: 100%;
}

.c3 {
  color: #5d9bec;
  font-size: 1rem;
  font-style: italic;
  line-height: 46px;
  width: 100%;
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="blueDark"
    font-size="medium"
    font-style="italic"
  >
    Você receberá:
  </h2>
  <ul
    class="c2"
  >
    <li
      class="c3"
    >
      Em 15 dias
      : 
      <strong>
        R$135.36
      </strong>
    </li>
    <li
      class="c3"
    >
      Em 30 dias
      : 
      <strong>
        R$138.24
      </strong>
    </li>
    <li
      class="c3"
    >
      Em 90 dias
      : 
      <strong>
        R$144.00
      </strong>
    </li>
  </ul>
</div>
`;
